src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
 function handleCardClick(name, link) {
   popUpImageModal.openModal(name, link);
 }
-
+//Popup for deleting cards from API along with handler
+const dltPopup = new PopupWithForm(constants.selectors.deleteSelector);
+dltPopup.setEventListeners();
+function handleDelete(id, card) {
+  dltPopup.openModal();
+  dltPopup.setSubmitAction(() => {
+    api
+      .deleteCards(id)
+      .then(() => {
+        dltPopup.closeModal();
+        card.deleteUI();
+      })
+      .finally(() => {
+        constants.dltFormSubmit.innerText = "Yes";
+      });
+  });
+}
+//Handler for the Like button
+function handleToggleLikes(id, isLiked, card) {
+  if (isLiked === false) {
+    card.handleLike();
+    isLiked = true;
+    api.toLike(id);
+  } else {
+    card.handleLike();
+    isLiked = false;
+    api.notLiked(id);
+  }
+}
 //FormValidation
-<<<<<<< Updated upstream
+//profile name/description
 const editFormValidator = new FormValidator(
-=======
-const editFormValidator = new FormValidator(settings, profileEditForm);
-editFormValidator.enableValidation();
-const addFormValidator = new FormValidator(settings, addModalForm);
-addFormValidator.enableValidation();
-
-const selectors = {
-  cardSection: ".gallery__cards",
-  cardTemplate: "#card-template",
-  popUpSelector: ".modal",
-  profileSelector: "#profile-edit-modal",
-  addSelector: "#add-modal",
-  imageSelector: ".gallery__card-image",
-  nameSelector: ".profile__name",
-  jobSelector: ".profile__description",
-  avatarSelector: ".profile__avatar",
-};
-const avatarFormValidator = new FormValidator(
->>>>>>> Stashed changes
   constants.settings,
   constants.profileEditForm
 );
-<<<<<<< Updated upstream
 editFormValidator.enableValidation();
+//avatar Form - link
+const avatarFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```